indico/modules/rb_new/util.py

```diff
@@ -219,6 +253,55 @@ def get_equipment_types():
     return [row.name for row in query]
 
 
+def get_blockings(room, candidates):
+    blocked_rooms = room.get_blocked_rooms(*(candidate.start_dt for candidate in candidates))
+    if blocked_rooms == []:
+        return {}
+    occurences = {}
+    dates = list(candidate.start_dt.date() for candidate in candidates)
+    for room in blocked_rooms:
+        blocking = room.blocking
+        for date in dates:
+            if blocking.start_date <= date <= blocking.end_date:
+                occurences[date] = [blocking]
+    return occurences
+
+
+def get_unbookable_hours(room):
+    hours = [hour for hour in room.bookable_hours]
+
+    if hours and len(hours) > 0:
+        inverted_hours = []
+        first = BookableHours(start_time=datetime.strptime('00:00', '%H:%M').time(), end_time=hours[0].start_time)
+        inverted_hours.append(first)
+        for i in range(1, len(hours)):
```

Comment:
Check if you can't use our `window()` util here:

    for foo, bar in window(hours):